src/eme.js

```diff
@@ -203,18 +224,26 @@ export const addSession = ({
       options,
       getLicense,
       removeSession,
+      getContentId,
       eventBus
     });
   }
 
-  video.pendingSessionData.push({
+  const sessionData = {
     initDataType,
     initData,
     options,
     getLicense,
     removeSession,
     eventBus
-  });
+  };
+
+  if (getContentId) {
+    sessionData.getContentId = getContentId;
+  }
```

Comment:
Since the `initData` shouldn't change, we may be better off getting the `contentId` early and potentially saving that to the session data instead of passing the function around and saving it.

Reply:
🤔